### PR TITLE
feat: Add Redux

### PR DESCRIPTION
**Pull request by lynnagara**
This PR adds Redux and refactors the `environment store` and `latest context store` from Reflux to Redux

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by dcramer** [comment]
Are we going to be removing Reflux as part of this? Otherwise we've got two (assumingely) huge flux frameworks we're running.

File: src/sentry/static/sentry/app/views/settings/account/avatar.jsx [hunk 1]
```diff
@@ -1,76 +1,74 @@
...
 import RadioGroup from '../components/forms/radioGroup';
 import LoadingError from '../../../components/loadingError';
 import LoadingIndicator from '../../../components/loadingIndicator';
 import {t} from '../../../locale';
 
-const AvatarSettings = createReactClass({
-  displayName: 'AvatarSettings',
-
-  propTypes: {
+class AvatarSettings extends React.Component {
```
Comment (dcramer) [inline src/sentry/static/sentry/app/views/settings/account/avatar.jsx]: aside, this kind of behavior is what AsyncView was really intended to solve: a component which needs to fetch data to render

**Comment by benvinegar** [comment]
Hey, I'm not opposed to this change, but I'd prefer we made it and committed ourselves to completely replacing Reflux after the current set of projects are completed. Otherwise we'll be left with Redux + Reflux together, probably for months, until a real plan is put in place.

**Comment by lynnagara** [comment]
Reflux doesn't seem to be supported or used much anymore, and I think Redux is the clear winner out of the flux-like frameworks.

I don't know about the other priorities, however I actually don't think doing a gradual migration is that terrible since:
- we get incremental benefits of being able to use Redux now
- it's the kind of thing that might never be a top priority
- bundle size is trivial

**Comment by benvinegar** [comment]
Let's discuss offline.

**Comment by EvanPurkhiser** [comment]
Removing app team on review, since this is stale at the moment.

**Comment by markstory** [comment]
Is this something we're still interested in pursuing or can it be closed because it can no longer be merged?

**Pull request closed without merging**